### PR TITLE
Allow separate build directory

### PR DESCRIPTION
**Pull request by dougnazar**
This series of patches enables compiling in a separate build directory from your source directory.
Also fixes the 3 different test harnesses.

It creates an identical install as `master` on my `Gentoo` systems.
It may need more work with different configurations, which I"ll be happy to work on if/when identified or suggested.

### CODE REVIEW

**Comment by sawolf** [comment]
Hey Doug - I could have sworn I left a comment on here earlier, but I guess not. Just wanted to say:
1) I agree that there's some value in having a separate build directory
2) Thanks for also fixing the tests
3) I've been putting this off relative to your other PRs just because it's going to take me a bit to review all of the changes. My intention is to get it handled for this coming release.

File: configure [hunk 1]
```diff
@@ -747,6 +747,7 @@ infodir
 docdir
 oldincludedir
 includedir
+runstatedir
```
Comment (sawolf) [inline configure:750]: All of these `runstatedir`-related changes get overwritten if you re-run `autoconf`

**Comment by sawolf** [comment]
In addition to the one note above, I tried to resolve the conflicts in `tap/configure` and `tap/ltmain.sh` and ran into bash syntax errors after running `./configure --enable-testing`. That one may be on me, though. 

If you intentionally made any changes that would end up in either of those files, I'd like to ask you to just make sure they make it into the relevant input files. `tap/configure` is built from `tap/configure.in`. I'm less certain about `tap/ltmain.sh` - that might be generated by autotools.

File: configure (hunk 1 above)
Comment (dougnazar) [reply configure:750]: Although I installed the older autoconf-2.69 on my system to minimize changes to the generated files, Gentoo does add a few patches which cause those differences. I'm fairly sure I checked that the extra changes didn't materially impact the installation. You can leave them in or regenerate the file on your reference platform.

**Comment by dougnazar** [comment]
The commit that changes the `tap` directory is labelled [Update libtap autoconf files to same version as nagios (2.69)](https://github.com/NagiosEnterprises/nagioscore/pull/867/commits/36ce09ea69394c55e2ce76577f3c4db7bed457b2). I don't remember changing anything there, I think it was just the `configure` scripts were too old and kept erroring out.

Again you should just be able to regenerate everything, or if you let me know the preferred dev environment I'll try to find some time to re-work the commits and as well as updating to HEAD.

**Comment by sawolf** [comment]
Thanks for the response!

I only picked out the `runstatedir` stuff because it looked like an intentional/human-generated change to me. If you're saying that was tool-generated, I don't think there's a problem. The VM I use to compile core is on CentOS Stream 8 + EPEL.

I'm fine with regenerating the files myself, I just wanted to make sure that I'm not destroying any of your work. Currently my intention is to pull this in after the coming release (tomorrow) and we'll put it in a 4.5.0 release next month.

**Comment by sawolf** [comment]
Took me a little longer than I'd like to get back around to merging this, but I just wanted to say thanks again for the patch, and we're anticipating a release in the next week or two.

**Pull request merged**